### ISSUE TITLE
Set up release plugin

### ISSUE THREAD

**Issue by GeorgiChochov** [AbsaOSS/enceladus#575]
## Background
Currently, we don't have a way to automatically handle our release process.

## Feature
We would like to be able to automatically create a `release/v{release-version}` branch, which once approved is merged into `master` without a `-SNAPSHOT` suffix, once in master tagged with the version. The `release` branch should also be merged into `develop` where the version should be incremented appropriately and suffxied with `-SNAPSHOT`.

## Proposed Solution
Solution Ideas 
1. Either use [gitflow-maven-plugin](https://github.com/aleksandr-m/gitflow-maven-plugin)
2. Or use [maven-release-plugin](https://maven.apache.org/maven-release/maven-release-plugin/)
